config/src/errors.rs

```diff
@@ -185,3 +185,46 @@ impl DsnpGraphError {
 		}
 	}
 }
+
+/// Macro to replicate Option<T>::ok_or, but logging if the returned
+/// Result is an Err variant.
+// (note: could have been implemented as a trait, but then the resulting log
+// event would not contain the correct file:line, since the std::file! macro records
+// the location only from an enclosing macro call)
+#[macro_export]
+macro_rules! ok_or_log {
```

Comment:
Yeah... I initially created it before I came up with the attribute macro approach... left it in because I thought it could still be useful in cases where we want to log internally to a function that doesn't return a `Result<>`

Reply:
BTW, if you want to see how this looks with the file + line information from the Java side, uncomment `showStandardStreams = true` in `build.gradle.kts` and run JUnit tests.

Reply:
Gonna leave it there; I think it will be useful for adding logs to places where the outer function doesn't bubble up the Result.